lock-api/src/main/java/com/palantir/lock/watch/ClientLockWatchEventLog.java

```diff
@@ -16,17 +16,12 @@
 
 package com.palantir.lock.watch;
 
-import java.util.Collection;
 import java.util.Optional;
-import java.util.Set;
 
-public interface LockWatchEventCache {
-    Optional<IdentifiedVersion> lastKnownVersion();
-    void processStartTransactionsUpdate(Set<Long> startTimestamps, LockWatchStateUpdate update);
-    void processGetCommitTimestampsUpdate(Collection<TransactionUpdate> transactionUpdates,
-            LockWatchStateUpdate update);
-    CommitUpdate getCommitUpdate(long startTs);
-    TransactionsLockWatchEvents getEventsForTransactions(Set<Long> startTimestamps,
-            Optional<IdentifiedVersion> version);
-    void removeTransactionStateFromCache(long startTimestamp);
+public interface ClientLockWatchEventLog {
+    ImmutableTransactionsLockWatchEvents.Builder getEventsBetweenVersions(
```

Comment:
The return type here feels wrong: why are you returning a builder?

Reply:
Reason is that we need to return
a) whether we had a snapshot or not;
b) a list of events

That's 2/3 of the return object at the cache level anyway. So we can either create a new type for this particular method, or return the builder for the cache to add the map and return it up. I'm open to going for the other route if you think that makes it cleaner.

Reply:
Let's not do it the way you have it right now, it rather breaks the usual mental model with builders and whatnot. And you introduce a rather hidden and weird dependency between them.

Reply:
Done

Reply:
Done